src/diffusers/loaders.py

```diff
@@ -151,7 +152,7 @@ def load_attn_procs(self, pretrained_model_name_or_path_or_dict: Union[str, Dict
         model_file = None
         if not isinstance(pretrained_model_name_or_path_or_dict, dict):
             # Let's first try to load .safetensors weights
-            if (is_safetensors_available() and weight_name is None) or (
+            if (use_safetensors is not False and weight_name is None) or (
```

Comment:
hmm, `is not False` is pretty confusing. With this and the above line,

```
        use_safetensors = kwargs.pop("use_safetensors", None if is_safetensors_available() else False)
```

I'm a bit confused about what the default behavior of the function is supposed to be

Reply:
Ah ok I see the comment later on in the PR. Could we change the code (here and in the other places where we take use_safetensors as an argument with a default) to approximately

```python
use_safetensors = kwargs.pop("use_safetensors", None)

if use_safetensors is None:
    use_safetensors = is_safetensors_available()

if use_safetensors and not is_safetensors_available():
  raise ValueError("`use_safetensors`=True but safetensors is not installed. Please install safetensors with `pip install safetenstors")

# in the rest of the function just use `use_safetensors` as a boolean

```

The rule of thumb here is that values which can be booleans or None can be hard to read/code with, especially when used in conditionals

Reply:
But that's not it here.

If `safetensors` is installed, but the weights do not exist on the remote, we don't want to fail, we want to use seamlessly the PT weights.

That's why it gets more complex.

False -> Use only PT (easy)
None -> Use safetensors if both available on system and remote (which we don't know when entering the function and without checking the remote)
True -> USe only safetensors (easy)

Reply:
Ah ok, I think the same logic still mostly applies

We can move the error check first and add another boolean for the case when we are allowed to fallback to non-safetensors

```py
use_safetensors = kwargs.pop("use_safetensors", None)


if use_safetensors and not is_safetensors_available():
  raise ValueError("`use_safetensors`=True but safetensors is not installed. Please install safetensors with `pip install safetenstors")

if use_safetensors is None:
    use_safetensors = is_safetensors_available()
    allow_pickle = True


# in the rest of the function just use `use_safetensors` as a boolean

# somewhere else
if use_safetensors and not_safetensors_compatible(...):
     if not allow_pickle:
         raise ValueError("... pickle not allowed ...")
     # do the pickle loading    

```

Is this correct or am I still missing something?

Reply:
2 booleans work, Not sure it's easier to read ( I to avoid variable that create nonsensical combos, for instance allow_pickle=True and use_safetensors=False doesn't make sense)

False 
None
True

Seems easier to reason about, but indeed creates some unpythoness.

Reply:
I'll move to 2 booleans if you think it's clearer

Reply:
t'would be much appreciated :)